app/src/main/java/com/simpragma/recipe/util/SearchItemActivity.java

```diff
@@ -0,0 +1,294 @@
+package com.simpragma.recipe.util;
+
+
+import android.arch.persistence.room.Room;
+import android.content.Context;
+import android.content.Intent;
+import android.content.pm.PackageInfo;
+import android.content.pm.PackageManager;
+import android.net.ConnectivityManager;
+import android.net.NetworkInfo;
+import android.net.Uri;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.support.v7.widget.GridLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.ProgressBar;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.android.volley.NetworkResponse;
+import com.android.volley.Request;
+import com.android.volley.RequestQueue;
+import com.android.volley.Response;
+import com.android.volley.VolleyError;
+import com.android.volley.toolbox.StringRequest;
+import com.android.volley.toolbox.Volley;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.simpragma.recipe.Constants;
+import com.simpragma.recipe.recipeapp.R;
+import com.simpragma.recipe.roomDatabase.AppDatabase;
+import com.simpragma.recipe.adapterclass.DataAdapter;
+import com.simpragma.recipe.pojoclass.Recipe;
+import com.simpragma.recipe.roomDatabase.RecipeRoomDB;
+import com.simpragma.recipe.pojoclass.RecipeList;
+import com.simpragma.recipe.roomDatabase.RoomDatabaseAdapter;
+
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class SearchItemActivity extends AppCompatActivity implements View.OnClickListener {
+
+    EditText searchEditText;
+    Button searchButton;
+    String TAG = getClass().getSimpleName(), url;
+    ArrayList<RecipeList> reciperesults;
+    ArrayList<RecipeList> addRecipelist;
+    Recipe post;
+    RecyclerView recyclerView;
+
+    GridLayoutManager gridLayoutManager;
+    ProgressBar progressBar;
+
+    private final int PAGE_START = 1;
+    private boolean isLoading = false;
+    private int TOTAL_PAGES = 3;
+    private int currentPage;
+
+    private boolean loading = true;
+    int pastVisiblesItems, visibleItemCount, totalItemCount;
+
+    private DataAdapter customAdapter;
+    TextView textView;
+    AppDatabase roomDB;
```

Comment:
These two lines are confusing. Need refactor

Reply:
TextView change it as errorMessageTextView.
AppDatabase change it as roomAdapterClass